tests/smoke-tests/test-randomness-consistency.ts

```diff
@@ -0,0 +1,142 @@
+import "@moonbeam-network/api-augment";
+import { ApiDecoration } from "@polkadot/api/types";
+import chalk from "chalk";
+import { expect } from "chai";
+import { printTokens } from "../util/logging";
+import { describeSmokeSuite } from "../util/setup-smoke-tests";
+
+// TEMPLATE: Remove useless types at the end
+import type { PalletProxyProxyDefinition } from "@polkadot/types/lookup";
+
+// TEMPLATE: Replace debug name
+const debug = require("debug")("smoke:randomness");
+
+const wssUrl = process.env.WSS_URL || null;
+const relayWssUrl = process.env.RELAY_WSS_URL || null;
+
+describeSmokeSuite(`Verify number of proxies per account`, { wssUrl, relayWssUrl }, (context) => {
+
+  let atBlockNumber: number = 0;
+  let apiAt: ApiDecoration<"promise"> = null;
+
+  const requestIds: number[] = [];
+  let requestCount: number = 0;
+
+  before("Retrieve all requests", async function () {
+    // It takes time to load all the proxies.
+    // TEMPLATE: Adapt the timeout to be an over-estimate
+    this.timeout(30_000); // 30s
+
+    const limit = 1000;
+    let last_key = "";
+    let count = 0;
+
+    atBlockNumber = process.env.BLOCK_NUMBER
+      ? parseInt(process.env.BLOCK_NUMBER)
+      : (await context.polkadotApi.rpc.chain.getHeader()).number.toNumber();
+    apiAt = await context.polkadotApi.at(
+      await context.polkadotApi.rpc.chain.getBlockHash(atBlockNumber)
+    );
+
+    // TEMPLATE: query the data
+    while (true) {
+      let query = await apiAt.query.randomness.requests.entriesPaged({
+        args: [],
+        pageSize: limit,
+        startKey: last_key,
+      });
+
+      if (query.length == 0) {
+        break;
+      }
+      count += query.length;
+
+      // TEMPLATE: convert the data into the format you want (usually a dictionary per account)
+      for (const request of query) {
+        const key = request[0].toString()
+
+        // requestId will be the last 8 bytes (16) nibbles but we need endianness swap
+        expect(key.length >= 18, "storage key should be at least 64 bits"); // assumes "0x"
+        const subkey = key.slice(-16);
+        let idHex = "";
+        for (let i = 0; i < subkey.length; i += 2) {
+          idHex += subkey.charAt(i+1);
+          idHex += subkey.charAt(i);
+        }
+        // reverse
+        idHex = idHex.split("").reverse().join("");
+        const requestId = parseInt(idHex, 16);
```

Comment:
@4meta5 here is where I'm able to decode the `RequestId` from the concat part of the hash